backend/e4h-services/ingestion-service/app/api/endpoints/file_ingestion.py

```
@@ -944,4 +967,379 @@ async def upload_facility_selection_excel_sheet(
+
+        # Save updated Excel
+        df.to_excel(excel_path, index=False, sheet_name=legacy_ticket_sheet_name)
+
+        return FileResponse(
+            path=excel_path,
+            filename=legacy_ticket_file.filename,
+            media_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+        )
+
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Error during duplicate check: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)